library/Fission/User/Query.hs

```diff
@@ -0,0 +1,17 @@
+module Fission.User.Query
+  ( getHerkouAddonByUUID
+  , getHerkouAddonByUserId
+  ) where
+
+import           Data.UUID (UUID)
+import           Database.Esqueleto hiding ((=.), update)
+import qualified Database.Persist as P
+
+import           Fission.Models
+import           Fission.Prelude
+
+getHerkouAddonByUserId :: MonadDB m => HerokuAddOnId -> Transaction m (Maybe (Entity User))
+getHerkouAddonByUserId addOnId = selectFirst [UserHerokuAddOnId P.==. (Just addOnId), UserActive P.==. True] []
+
+getHerkouAddonByUUID :: MonadDB m => UUID -> Transaction m (Maybe (Entity HerokuAddOn))
+getHerkouAddonByUUID uuid = selectFirst [HerokuAddOnUuid P.==. uuid] []
```

Comment:
Does naming a function get us much over just using `selectFirst`?

Other than that:

![](https://media3.giphy.com/media/3o8doT9BL7dgtolp7O/giphy.gif?cid=36b14face72fca6282c311de49383a763dbad80aac91e844&rid=giphy.gif)

Reply:
So this is one convention ive picked up where I like to isolate anything that uses sql like conventions to a file under named functions to start building up our DSL.

So while its not much I think it improves both code readability and isolation of the database logic.

Reply:
>  isolation of the database logic.

We get a lot of that from the types and classes. But yes, in general I like this pattern. We used it pretty extensively in Lonely Planet for example, IIRC. Part of that was a consequence of Ecto letting you do _whatever_, but yeah, can't hurt to add some abstraction 👍